src/dhtproto/client/legacy/internal/helper/ExtensibleChannelMirror.d

```diff
@@ -59,13 +59,15 @@ import dhtproto.client.DhtClient;
     Template_Params:
         Dht = type of dht client (must be derived from DhtClient and contain the
             RequestScheduler plugin)
+        MirrorImpl = type of channel mirror implementation; must be derived from
+            MirrorBase!(Dht)
         Plugins = variadic list of record-handling plugins to be called (*in the
             order specified*) when records are received by the mirror
 
 *******************************************************************************/
 
-public class ExtensibleChannelMirror ( Dht : DhtClient, Plugins ... )
-    : ChannelMirror!(Dht)
+public class ExtensibleMirror
+    ( Dht : DhtClient, MirrorImpl : MirrorBase!(Dht), Plugins ... ) : MirrorImpl
```

Comment:
I wonder if you could reasonably avoid the boilerplate-y requirement to specify both `Dht` and `MirrorImpl` here.  The `MirrorImpl` comes with the DHT client type defined, so there must be some other means to validate that it is indeed a subclass of `MirrorBase` ... ?

Reply:
Yeah, good point. I'll have a look. It might be possible.

Reply:
Minor gripe, of course.  The new setup seems to be working very nicely in general.

Reply:
I can't see a simple way of doing this. I'll leave it as it is; I don't think it's worth spending time reworking stuff.

Reply:
No problem.  It's a small price to pay for a really nice improvement in how legacy mirrors work.